physical/postgresql.go

```diff
@@ -175,3 +199,143 @@ func (m *PostgreSQLBackend) List(prefix string) ([]string, error) {
 
 	return keys, nil
 }
+
+func (m *PostgreSQLBackend) HAEnabled() bool {
+	return true
+}
+
+// PostgreSQLLock implements the Lock interface for PostgreSQL
+type PostgreSQLLock struct {
+	client *sql.DB
+	table  string
+	leader chan struct{}
+
+	key     string
+	value   string
+	vaultID string
+}
+
+// LockWith initializes a Postgres backend lock
+func (m *PostgreSQLBackend) LockWith(key, value string) (Lock, error) {
+	id, err := uuid.GenerateUUID()
+	if err != nil {
+		return nil, err
+	}
+	return &PostgreSQLLock{
+		client:  m.client,
+		table:   m.lock_table,
+		leader:  make(chan struct{}),
+		key:     key,
+		value:   value,
+		vaultID: id,
+	}, nil
+}
+
+// Lock grabs a lock, or waits until it is available
+func (m *PostgreSQLLock) Lock(stopCh <-chan struct{}) (<-chan struct{}, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		<-stopCh
+		cancel()
+	}()
+	ticker := time.NewTicker(PostgreSQLLockRetryInterval)
+	defer ticker.Stop()
+	for {
+		_, err := m.client.ExecContext(
+			ctx,
+			"DELETE FROM "+m.table+" WHERE expiration < now()",
+		)
+		if err != nil {
+			return nil, err
+		}
+		lockTTL := strconv.Itoa(PostgreSQLLockTTL)
```

Comment:
Remove `lockTTL`.

Add a small helper function, `relationName()`:

```go
func (m *PostgreSQLLock) relationName() string
  relationName := pq.QuoteIdentifier(m.lockTableName)
  if schemaName == "" {
    relationName = fmt.Sprintf("%s.%s", pq.QuoteIdentifier(m.lockSchemaName), pq.QuoteIdentifier(m.lockTableName))
  }
  return relationName
}
```

Both `lockTableName` and `lockSchemaName` were already `pq.QuoteIdentifier()`'ed, but if this gets moved to a helper function then we can remove the `pq.QuoteIdentifer()` at config time.  This is defensive in advance of a HUP signal handler.